nengo_gui/static/components/spa_similarity.js

```diff
@@ -42,10 +42,10 @@ Nengo.SpaSimilarity.prototype.reset_legend_and_data = function(new_labels){
     while(this.legend.lastChild){
         this.legend.removeChild(this.legend.lastChild);
     }
-    this.legend_svg = d3.select(this.legend).append("svg");
+    this.legend_svg = d3.select(this.legend).append("svg").attr("id", "id"+this.uid);
```

Comment:
I'm uncomfortable with the ```id``` of the legend being set to ```"id<id-of-base-component>"```.  Basically, I'm worried about someone in the future coming along and needing to set an id for the basic component and using ```"id<id-of-base-component>"```, resulting in multiple objects with the same id.  I'd be happier if the ```id``` here was changed to ```"legend<id-of-base-component>"```.